clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -947,12 +947,19 @@ run_lockstep_simulation(scheduler_t &scheduler, int num_outputs, memref_tid_t ti
                 eof[i] = true;
                 continue;
             }
-            if (status == scheduler_t::STATUS_WAIT)
+            if (status == scheduler_t::STATUS_WAIT) {
+                sched_as_string[i] += '-';
                 continue;
+            }
             assert(status == scheduler_t::STATUS_OK);
             if (type_is_instr(memref.instr.type)) {
                 sched_as_string[i] +=
                     'A' + static_cast<char>(memref.instr.tid - tid_base);
+            } else {
+                // While this makes the string longer, it is just too confusing
```

Comment:
Now that we have this, can we bring back the assertion on the same input not being on multiple cores, which we removed in https://github.com/DynamoRIO/dynamorio/commit/2cc331a790d18c4b57c04411e97353f3c8ed15ae.

Reply:
Added inside the lockstep routine so it will apply to every test using that.